components/dashboard/src/projects/ConfigureProject.tsx

```diff
@@ -54,145 +56,199 @@ const TASKS = {
-      if (repoConfigString) {
-        setIsDetecting(false);
-        setEditorMessage(<EditorMessage type="warning" heading="Configuration already exists in git." message="Run a prebuild or open a new workspace to edit project configuration."/>);
-        setGitpodYml(repoConfigString);
-        return;
-      }
-      if (project.config && project.config['.gitpod.yml']) {
+    const { teams } = useContext(TeamsContext);
+    const location = useLocation();
+    const team = getCurrentTeam(location, teams);
+    const routeMatch = useRouteMatch<{ teamSlug: string, projectSlug: string }>("/(t/)?:teamSlug/:projectSlug/configure");
+    const [project, setProject] = useState<Project | undefined>();
+    const [gitpodYml, setGitpodYml] = useState<string>('');
+    const [dockerfile, setDockerfile] = useState<string>('');
+    const [editorMessage, setEditorMessage] = useState<React.ReactNode | null>(null);
+    const [selectedEditor, setSelectedEditor] = useState<'.gitpod.yml' | '.gitpod.Dockerfile'>('.gitpod.yml');
+    const [isEditorDisabled, setIsEditorDisabled] = useState<boolean>(true);
+    const [isDetecting, setIsDetecting] = useState<boolean>(true);
+    const [prebuildWasTriggered, setPrebuildWasTriggered] = useState<boolean>(false);
+    const [startPrebuildResult, setStartPrebuildResult] = useState<StartPrebuildResult | undefined>();
+    const [prebuildInstance, setPrebuildInstance] = useState<WorkspaceInstance | undefined>();
+    const { isDark } = useContext(ThemeContext);
+
+    const [showAuthBanner, setShowAuthBanner] = useState<{ host: string } | undefined>(undefined);
+
+    useEffect(() => {
+        // Disable editing while loading, or when the config comes from Git.
+        setIsDetecting(true);
+        setIsEditorDisabled(true);
+        setEditorMessage(null);
+        if (!teams) {
+            setIsDetecting(false);
+            setEditorMessage(<EditorMessage type="warning" heading="Couldn't load teams information." message="Please try to reload this page." />);
+            return;
+        }
+        (async () => {
+            const projects = (!!team
+                ? await getGitpodService().server.getTeamProjects(team.id)
+                : await getGitpodService().server.getUserProjects());
+            const project = projects.find(p => p.name === routeMatch?.params.projectSlug);
+            if (!project) {
+                setIsDetecting(false);
+                setEditorMessage(<EditorMessage type="warning" heading="Couldn't load project information." message="Please try to reload this page." />);
+                return;
+            }
+            setProject(project);
+            try {
+                await updateProjectConfiguration(project);
+            } catch (error) {
+                if (error && error.code === ErrorCodes.NOT_AUTHENTICATED) {
+                    setShowAuthBanner({ host: new URL(project.cloneUrl).hostname });
+                }
```

Comment:
Nit: Please don't sweep other errors silently under the carpet 😇 

```suggestion
                } else {
                    console.error('Getting project configuration failed', error);
                }
```